module/zfs/spa_misc.c

```diff
@@ -2245,12 +2245,6 @@ spa_name_compare(const void *a1, const void *a2)
 	return (AVL_ISIGN(s));
 }
 
-void
-spa_boot_init(void)
-{
-	spa_config_load();
-}
```

Comment:
FreeBSD makes use of this hook so it was preserved originally, even though it's unused on Linux.  That said, I'm not opposed to removing it now and it can be added back in the needed form for FreeBSD when those changes are integrated.

Reply:
Brought this back.